libs/lib-linker/src/lib/consolidation-result/consolidation-result.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, Input, OnChanges } from '@angular/core';
+import { LinkerResult } from '../lib-linker';
+
+@Component({
+  selector: 'perun-web-apps-consolidation-result',
+  templateUrl: './consolidation-result.component.html',
+  styleUrls: ['./consolidation-result.component.scss'],
+})
+export class ConsolidationResultComponent implements OnChanges {
+  @Input()
+  result: LinkerResult;
+  color = '';
+  icon = '';
+
+  greenColor = '#e0ffd4';
+  blueColor = '#d4f2ff';
+  redColor = '#ffd4d4';
+  warningIcon = 'warning';
+  infoIcon = 'info';
+  messageSent = 'mark_email_read';
+
+  ngOnChanges(): void {
+    switch (this.result) {
```

Comment:
I would prefer if the `result` input would be a setter and this logic would be inside of it. Since this only needs to be evaluated when the result changes not on every change.

Reply:
So the only input this component have is `this.result` so the `ngOnChanges` method will happen when the result changes which is something what we want

Reply:
Yes, you are technically correct, as long as no other input is added.